File: CodeForces/501-600/546/A/546A.cs
// Problem 546A: Soldier and Bananas
// https://codeforces.com/contest/546/problem/A
// Compiled with: Mono JIT compiler version 6.8.0.105 (Debian 6.8.0.105+dfsg-3.3+deb12u1 Sat Jun 21 16:33:59 UTC 2025)
// Linux version: Linux kernel 6.1.0-10-amd64
// Submitted on: October 21st, 2025
// Codeforces language used: C# Mono 6.8

using System;

class Program {
    static void Main() {
        string all_input = Console.ReadLine();
        string[] parts = all_input.Split(' ');

        int k = int.Parse(parts[0]);
        int n = int.Parse(parts[1]);
        int w = int.Parse(parts[2]);

        int total_cost_multi = ((w*(w+1))/2);
        int total_cost = k * total_cost_multi;

        Console.WriteLine(Math.Max(total_cost-n, 0));
    }
}

File: CodeForces/501-600/546/A/546A.pl
#!/usr/bin/perl

#! Problem 546A: Soldier and Bananas
#! https://codeforces.com/contest/546/problem/A
#! Executed with: perl 5, version 36, subversion 0 (v5.36.0) built for x86_64-linux-gnu-thread-multi
#! Linux version: Linux kernel 6.1.0-10-amd64
#! Submitted on: October 24th, 2025
#! Codeforces language used: Perl 5.20.1

use strict;
use warnings;


my @arr = split(' ', <STDIN> =~ s/\n$//r);

my $k = $arr[0];
my $n = $arr[1];
my $w = $arr[2];

my $cost_multiplier = (($w*($w+1))/2);
my $total_cost = $k * $cost_multiplier;
my $to_borrow = $total_cost - $n;

if ($to_borrow < 0) {
    print "0\n";
} else {
    print "$to_borrow\n";
}
